pubsub/envelope.go

```diff
@@ -110,6 +116,29 @@ func NewCloudEventsEnvelope(id, source, eventType, subject string, topic string,
 	}
 }
 
+// HasExpired determines if the current cloud event has expired.
+func (cloudEvent *CloudEventsEnvelope) HasExpired() bool {
+	if cloudEvent.Expiration != "" {
+		expiration, err := time.Parse(time.RFC3339, cloudEvent.Expiration)
+		if err != nil {
+			return false
+		}
+
+		return expiration.UTC().Before(time.Now().UTC())
+	}
+
+	return false
+}
```

Comment:
We are not implementing deadletter queue in Dapr yet, just TTL. I like the idea of Dapr not doing any manual expiration if component supports it natively. I wonder what would be the best way to implement that. It would be a change in how Dapr implements building blocks, where the common logic in Dapr changes based on the component.

Reply:
> It would be a change in how Dapr implements building blocks, where the common logic in Dapr changes based on the component.

We are already doing this with Bulk Get. If a component natively supports a Bulk Get operation, the component takes over. if not, it signals to Dapr that it doesn't support it and then Dapr operates it's own logic.

This would be the same here: if the component natively supports TTL it would do it, and if not it would signal to Dapr it doesn't have this capability and so Dapr will expire the message when receiving it.

Reply:
OK. I like this approach.

Reply:
This logic has been implemented now. Dapr will not do any expiration if component implementation handles that.